ui/src/app/(index)/page.tsx

```diff
@@ -1,7 +1,79 @@
+import React from 'react';
+import Announcement from '@/components/home/Announcement';
+
+const Home = async () => {
+    const [currentUser, numberOfGroupings, numberOfMemberships, announcements] = await Promise.all([
+        getCurrentUser(),
+        getNumberOfGroupings(),
+        getNumberOfMemberships(),
+        getAnnouncements()
+    ]);
+
+    const handleAnnouncements = () => {
+        if (Object.prototype.hasOwnProperty.call(announcements, 'announcements')) {
+            return announcements.announcements
+                .filter((announcement) => announcement.state !== 'Active')
+                .map((announcement) => announcement.message);
+        } else {
+            return [];
+        }
+    };
+
     return (
-        null
+        <main>
+            <div className="container mt-5 mb-5">
+                {handleAnnouncements().map((announcement: string, index: number) => (
+                    <div key={index}>
+                        <Announcement announcement={announcement}/>
+                    </div>
+                ))}
+                <div className="flex flex-row py-8 px-0.5 justify-between">
+                    <div className="md:w-7/12 flex items-center">
+                        <div>
+                            <h1 className="sr-only">UH Groupings</h1>
+                            <Image
+                                src="uhgroupings/uh-groupings-text.svg"
+                                alt="UH Groupings logotype"
+                                width={337.5}
+                                height={52.5}
+                                className="w-[210px] h-[32.67px] sm:w-1/2 sm:h-auto md:w-[337.5px] md:h-[52.5px]"
+                            />
+
+                            <p className="text-xl mt-1"> Manage your groupings in one place, use them in many.</p>
+                            <div className="mt-4">
+                                {currentUser ? (
+                                    <LoginButton currentUser={currentUser}/>
+                                ) : null}
+                            </div>
+                        </div>
+                    </div>
+                    <div className="hidden lg:block md:w-5/12">
+                        <Image
+                            src="/uhgroupings/uh-groupings-logo-large.svg"
+                            alt="UH Groupings"
+                            width={365.5}
+                            height={292.5}
+                        />
+                    </div>
+                </div>
+            </div>
+
+            {currentUser && currentUser.roles.includes(Role.UH) ? (
+                <AfterLogin
+                    currentUser={currentUser}
+                    numberOfGroupings={numberOfGroupings}
+                    numberOfMemberships={numberOfMemberships}/>
+            ) : (
+                <BeforeLogin/>
+            )}
```

Comment:
You do not need the `currentUser &&` part here, `currentUser.roles.includes(Role.UH)` is enough. Thanks!

Reply:
done